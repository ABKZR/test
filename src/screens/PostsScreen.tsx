import React from 'react';
import {View, StyleSheet, SafeAreaView} from 'react-native';
import usePosts from '../hooks/usePosts';
import Loader from '../components/Loader';
import FlatListHandler from '../components/FlatListHandler';
import ErrorView from '../components/ErrorView';

const PostsScreen: React.FC = () => {
  const {data, status, loadMore, handleRefresh} = usePosts();

  if (status === 'failed') {
    return <ErrorView onRetry={handleRefresh} />;
  }

  return (
    <SafeAreaView style={styles.container}>
      <View>
        {status === 'loading' && !data.length ? (
          <Loader />
        ) : (
          <FlatListHandler
            data={data}
            onLoadMore={loadMore}
            onRefresh={handleRefresh}
            refreshing={status === 'loading'}
          />
        )}
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#ffffff',
  },
});

export default PostsScreen;
