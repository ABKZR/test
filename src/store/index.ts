import { createStore, applyMiddleware, combineReducers } from 'redux';
import { thunk as thunkMiddleware } from 'redux-thunk';
import postsReducer from './posts/reducer';

const rootReducer = combineReducers({
  posts: postsReducer,
});

export type RootState = ReturnType<typeof rootReducer>;

export const store = createStore(rootReducer, applyMiddleware(thunkMiddleware));
