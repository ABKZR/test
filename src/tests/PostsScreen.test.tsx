import React from 'react';
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import { store } from '../store';
import PostsScreen from '../screens/PostsScreen';

describe('PostsScreen Snapshot', () => {
  it('renders correctly', () => {
    const tree = render(
      <Provider store={store}>
        <PostsScreen />
      </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
