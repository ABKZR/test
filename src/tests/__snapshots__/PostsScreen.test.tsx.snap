// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostsScreen Snapshot renders correctly 1`] = `
<View
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      size="large"
    />
  </View>
</View>
`;
